### DISCUSSION TITLE
Pokedex. Extraer más de un dato de un API

### DISCUSSION THREAD

**Discussion by Juan-2003** [LaunchX-InnovaccionVirtual/FrontEnd-Mision#475] [Q&A]
Para la pokedex, para la sección de movimientps, hay manera de extraer todos los movimientos sin necesidad de copiar y pegar la misma linea de código? 
Ahorita lo estoy haciendo asi, pero me gustaría extraer todos los movimientos de una.
![image](https://user-images.githubusercontent.com/99064991/159191387-3fe1ae47-ab1d-43f0-8c30-ae36502931b6.png)

**Comment by AldairRoman** [comment]
Puedes guardar todos los movimientos como let movimientos=data.moves y en una función pasarle movimientos  y luego dentro de  la función lo recorres con un foreach o con un for  como si lo hicieras manual con una variable i por ejemplo hasta que llegue el tamaño de movimientos . 
Espero haberme dado a entender.


Saludos

**Comment by richirrim** [comment]
Claro, imagino que todo los movimientos están dentro de `moves` y que `moves` es un array y sus elementos objetos verdad? Si es así, puedes hacerlo iterando sobre el array `moves` con un `for...of`.

Algo así:
 
```js
const moves = data.moves

// Y así podrás obtener todos.
for (let move of moves) {
    // revisas la consola y verás el resultado.
    console.log(move.name) 
}
```

**Reply by AndreaPinedaM** [reply]
ya lo intenté y sale " 'x' (no. de movimientos) undefined". Igual ya intenté hacerlo con un for tomando como parámetro la longitud del arreglo pero tampoco me sale. ):

**Reply by richirrim** [reply]
Creo que estás capturando mal lo que te devuelve la API, usa consolé.log para que veas que te dévuelve cada línea de tu código, talvez y así te das cuenta porq no está funcionando cuando debería de funcionar correctamente el código.

Estás usando async/await? Recuerda q las peticiones a la API son asíncronas y si no usas el await podrías no obtener el resultado que esperas.

**Comment by AngelCruzO** [comment]
Ocupas un for con iterador:
let movimientos;
for(var i=0;i<data.moves.length;i++{
   movimientos += data.moves[i].move.name;
}

console.log(movimientos)

**Comment by sergiobps** [comment]
Yo intente como dicen antes y me devolvía "**undefined**", busque en San Google y encontré el Object.keys que devuelve la cantidad de elementos que tiene un objeto.
```
        let pokeMovs = "";
        let cantMovs = Object.keys(data.moves).length;
        for(let i = 0; i < cantMovs ; i++)
        {
            pokeMovs += data.moves[i].move.name + "\n";
        }

```

**Comment by dalexez** [comment]
Hola! yo me encontré con el mismo problema y lo que se me hiso mas fácil fue usar el método **map()**, este método te crea un nuevo array iterando por cada uno de los elementos del array original, en mi caso lo use por ejemplo para los tipos de Pokémon que en pueden tener 1 o 2 tipos.

![image](https://user-images.githubusercontent.com/62618610/159352193-b5fc57f8-7410-4a31-92da-0117f20aede7.png)

Usando lo de la imagen de arriba, digamos que busco a **"bulbasaur"** tiene **2 typos de pokemon**, me regresa esto:

![image](https://user-images.githubusercontent.com/62618610/159352248-3baf28e3-a0a1-44c2-8160-3671a4488aed.png)

Y si consulto un pokemon con solo un tipo como **"charmander"** me regresa esto:

![image](https://user-images.githubusercontent.com/62618610/159352848-965919d6-2dfd-420c-96a8-f3faa4b76082.png)

Quizá te pueda servir para extraer todos los movimientos y guardarlos en un arreglo con una solo linea de codigo **(buenos como 2 jeje)**, posteriormente usar los datos 
Saludos!!

**Comment by D-CIoud** [comment]
Hola buen día mira lo que yo hice pare esto, fue crear una lista para guardar todos los movimientos, los guarde en una variable y después con un foreach los concatene para meterlos en una sola variable y posteriormente imprimirla dentro de su respectivo input.

![image](https://user-images.githubusercontent.com/36202151/159358162-71a36223-a69b-4d29-a4c4-442278ab3931.png)

Esto es lo que obtengo para mostrar:
![image](https://user-images.githubusercontent.com/36202151/159358259-ea63acc8-9574-44d9-8fd7-8e5a1a9c0eba.png)

**Reply by MarceloCortesC** [reply]
y para las caracteristicas hiciste lo misma con cada una?
